crates/telio-proxy/src/proxy.rs

```diff
@@ -181,10 +195,27 @@ impl UdpProxy {
         let _ = self.task_egress.stop().await.resume_unwind();
         let _ = self.task_ingress.stop().await.resume_unwind();
     }
+
+    ///
+    pub async fn on_network_change(&self) {
+        let _ = task_exec!(&self.task_egress, async move |state| {
+            state.num_socket_changes = 0;
```

Comment:
Any reason not to reset this on reconfiguration?

Also if we have more then 3 peers, wouldn't a singular no net allow just two be recreated, leaving rest stuck? as these changes are global?

Reply:
No reason, just a hasty oversight. Fixed that, and updated the socket limit to be per public key, instead of a single global one